File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        
    	int[] answer_array = new int[nums.length];
    	
    	for(int i=0; i < answer_array.length; i++) {
    		answer_array[i] = 1;
    	}
    	
    	for (int i=1; i < nums.length; i++) {
    		
    		answer_array[i] = answer_array[i-1] * nums[i-1];
    		
    	}
    	
    	int right_product = 1;
    	
    	for (int i = (nums.length)-1; i >= 0 ; i--) {
    		
    		answer_array[i] = answer_array[i] * right_product;
    		
    		right_product = right_product * nums[i];
    	}
    	
    	return answer_array;	
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        answer_list = [1] * len(nums)

        for i in range(1, len(nums)):
            answer_list[i] = answer_list[i-1]  * nums[i-1]

        right_product = 1
        for i in range(len(nums)-1, -1, -1):
            answer_list[i] = answer_list[i] * right_product
            right_product = right_product * nums[i]

        return answer_list